packages/transport/src/transports/abstractUsb.ts

```diff
@@ -85,56 +85,55 @@ export abstract class AbstractUsbTransport extends AbstractTransport {
     }
 
     public acquire({ input }: { input: AcquireInput }) {
-        return this.scheduleAction(async () => {
-            const { path } = input;
+        return this.scheduleAction(
+            async () => {
+                const { path } = input;
+
+                if (this.listening) {
+                    this.listenPromise[path] = createDeferred<string>();
+                    this.listenPromise[path].promise
+                        .catch(err => err)
+                        .finally(() => {
+                            delete this.listenPromise[path];
+                        });
+                }
 
-            if (this.listening) {
-                this.listenPromise[path] = createDeferred<string>();
-            }
+                const acquireIntentResponse = await this.sessionsClient.acquireIntent(input);
 
-            const acquireIntentResponse = await this.sessionsClient.acquireIntent(input);
-            if (!acquireIntentResponse.success) {
-                if (this.acquirePromise) {
-                    this.acquirePromise.resolve(undefined);
+                if (!acquireIntentResponse.success) {
+                    if (this.acquirePromise) {
+                        this.acquirePromise.resolve(undefined);
+                    }
+                    return this.error({ error: acquireIntentResponse.error });
                 }
-                return this.error({ error: acquireIntentResponse.error });
-            }
-            this.acquiredUnconfirmed[path] = acquireIntentResponse.payload.session;
 
-            const reset = !!input.previous;
-            const openDeviceResult = await this.transportInterface.openDevice(path, reset);
-            if (!openDeviceResult.success) {
-                if (this.listenPromise) {
-                    this.listenPromise[path].reject(new Error(openDeviceResult.error));
+                this.acquiredUnconfirmed[path] = acquireIntentResponse.payload.session;
+
+                const reset = !!input.previous;
+                const openDeviceResult = await this.transportInterface.openDevice(path, reset);
+
+                if (!openDeviceResult.success) {
+                    if (this.listenPromise) {
+                        this.listenPromise[path].reject(new Error(openDeviceResult.error));
```

Comment:
I'm not sure, but couldn't this clean itself in the `finally` block before the end of this method and therefore return success instead of the promise? Are we dependent on the execution order?